pgmpy/prediction/NaiveBackdoorRegressor.py

```
@@ -0,0 +1,284 @@
+        missing_columns = set(required_features) - set(X_df.columns)
+        if missing_columns:
+            raise ValueError(
+                f"Missing required columns in input data: {list(missing_columns)}. "
+                f"Required columns based on DAG roles: {required_features}."
+            )
+
+        return X_df[required_features]
+
+    def fit(self, X, y, sample_weight: Optional[np.ndarray] = None, feature_names=None):
+        """
+        Fit the Naive Backdoor Regressor.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or pandas DataFrame
+            Training data. If DataFrame, must include columns for exposure,
+            adjustment, and pretreatment variables as defined in causal graph roles.
+            If array, `feature_names` must be provided unless used within a
+            generic sklearn context (like compatibility tests).
+        y : array-like of shape (n_samples,)
+            Target values (outcome variable).
+        sample_weight : array-like of shape (n_samples,), optional
+            Sample weights.
+        feature_names : list, optional
+            Feature names when X is an array. Strongly recommended for array inputs
+            to ensure correct semantic mapping.
+
+        Returns
+        -------
+        self
+            Returns self for method chaining.
+        """
+        X_arr, y_arr = check_X_y(
+            X, y, accept_sparse=False, ensure_2d=True, dtype="numeric"
+        )
+
+        # Extract and validate causal graph roles and define required features
+        exposure_var, outcome_var, adjustment_vars, pretreatment_vars = (
+            self._validate_dag_and_extract_roles()
+        )
+        self.exposure_var_ = exposure_var
+        self.outcome_var_ = outcome_var
+        self.adjustment_vars_ = adjustment_vars
+        self.pretreatment_vars_ = pretreatment_vars
+        self.feature_columns_ = self._extract_feature_columns()
```

Shouldn't we do this in the `__init__` method, as all these variables are defined in the `causal_graph` itself? I think it is better to fail in `__init__` if something is missing in the model rather than waiting for the `fit` call. I don't know if this has something to do with sklearn compatibility? @fkiraly @marrov Thoughts?

---

Update: I looked at the sklearn docs, and these should actually remain here. Ignore my previous comment.